src/veins/modules/mobility/traci/TraCIMobility.cc

```diff
@@ -296,6 +304,32 @@ double TraCIMobility::calculateCO2emission(double v, double a) const
     double zeta = 0.241;
     double alpha1 = 0.973;
 
+    // Eq 9 - please note: gamma was dropped in comparison to Eq 6
+    if (P_tract <= 0) return alpha1;
+    return alpha + beta * v * 3.6 + delta * v * v * v * (3.6 * 3.6 * 3.6) + zeta * a * v;
+}
+
+double TraCIMobility::calculateFuelConsumption(double v, double a) const
```

Comment:
If I understand correctly, the output describes the rate of fuel consumption, maybe the function's name could be adapted to reflect that.